src/variables.cpp

```diff
@@ -886,8 +886,16 @@ bool VarValue::ValueEqual( VarValue &rhs )
 					for (size_t c = 0; c < m_val.ncols(); c++)
 					  {
 						if (equal)
-//							equal = equal && (m_val(r, c) == rhs.m_val(r, c));
-							equal = equal && (RelDif(m_val(r, c),rhs.m_val(r, c)) < TOLERANCE);
+						{
+							double x = m_val(r, c);
+							double y = rhs.m_val(r, c);
+							if ((std::isnan(x)) || (std::isnan(y)))
+								equal = equal && ((std::isnan(x)) && (std::isnan(y)));
+							else if ((std::isinf(x)) || (std::isinf(y)))
+								equal = equal && ((std::isinf(x)) && (std::isinf(y)));
+							else
+								equal = equal && (RelDif(m_val(r, c), rhs.m_val(r, c)) < TOLERANCE);
+						}
```

Comment:
This makes more sense, nice!

I think `equal = equal && <>` can be replaced with simply `equal = <>` since we're in an if-statement ensuring that equal = 1 and `1 && x = x`. 

Then after `else` if not equal, you can break to cut the iteration short.